### DISCUSSION TITLE
Jupyter Notebook can't open.

### DISCUSSION THREAD

**Discussion by scottbarksdale** [sqlitebrowser/sqlitebrowser#3729] [General]
On attempting to open a Jupyter Notebook file, get the following message: 500:Internal Server Error
Open the "my_database.db" file and get this message: Error! /Users/scottsmac/my_database not UTF-8 encoded

**Comment by FriedrichFroebel** [comment]
Are you sure you are talking about DB4S? DB4S cannot open Jupyter Notebook files and `my_database.db` does not seem to be a SQLite3 database file, but a plaintext one. Are you trying to open a file with SQL statements within DB4S?

**Reply by scottbarksdale** [reply]
"About DB Browser for SQLite"

DB Browser for SQLite Version 3.12.2

Built for x86_64-little_endian-lp64, running on x86_64
Qt Version 5.15.2
SQLCipher Version 4.4.3 community (based on SQLite 3.34.1).

here's some of the code 

import pandas as pd
from IPython.display import display
import sqlite3
import seaborn as sns

# Load the CSV data
df = pd.read_csv('Census_Data.csv')

# Create a connection to an in-memory SQLite database
conn = sqlite3.connect(':memory:')

#additional code below...not shown

**Reply by FriedrichFroebel** [reply]
I still do not get how DB4S/sqlitebrowser is related to your Python code/Jupyter Notebooks. Could you please elaborate?

**Comment by scottbarksdale** [comment]
I'm trying to improve my SQL skills and using python and sql within a jupyter notebook environment. I've uploaded a csv file to the environment which I then convert the csv file with pandas dataframe. Then converting the dataframe to a table so that I can then query the table with sql code. Evidently the table is stored in a sqlite database as I include SQLite3 within the code. Please let me know if this makes sense. Thanks

**Reply by FriedrichFroebel** [reply]
So you are generating a SQLite3 database from within Python code. What is the problem now within DB4S? Aren't you able to open the generated database with it? Or is there another issue?
